src/replay/service.zig

```diff
@@ -28,100 +28,146 @@ const EpochTracker = replay.trackers.EpochTracker;
 const SlotTracker = replay.trackers.SlotTracker;
 const SlotTree = replay.trackers.SlotTree;
 
+const GossipVerifiedVoteHash = sig.consensus.vote_listener.GossipVerifiedVoteHash;
+const ThresholdConfirmedSlot = sig.consensus.vote_listener.ThresholdConfirmedSlot;
+
 const updateSysvarsForNewSlot = replay.update_sysvar.updateSysvarsForNewSlot;
 
 pub const Logger = sig.trace.Logger("replay");
 
-pub const Service = struct {
-    replay: ReplayState,
-    consensus: ?TowerConsensus,
-    num_threads: u32,
+pub const AdvanceReplayParams = struct {
+    want_multi_threaded_replay: bool,
+    replay_state: *ReplayState,
+    consensus: ?struct {
+        tower: *TowerConsensus,
```

Comment:
Why did you make ReplayState and TowerConsensus pointers? It's more state to juggle in whatever context needs to create this struct. I liked how this was all owned by one entity.

Reply:
Having it all owned by one entity was significantly more complicated due to the layered-ness of the optionals, both in terms of testing the relevant components, and the fact that it obligated the use of much more layered and complicated ownership rules, which ties in to the removal of the `Owned` wrappers; without all these layers of ownership, how data is supposed to be handled becomes much simpler to reason about.

Reply:
I see that it's more flexible but it adds more layers that make the code more confusing to read. I don't see how the flexibility is actually being taken advantage of in this scenario. If it's just for testing, I'd rather not clutter the production code with extra layers just to make testing a little more convenient.

I'm not sure I see how there are any fewer embedded abstractions. Are you talking about removing `Owned`? The abstraction and semantics that it represented are still present. They're just explained with a comment instead of being enforced by a type, which I think actually harms clarity by giving the illusion that the semantics are simpler than they really are.